### PR TITLE
New package: xmlsec Fixes #2404

### LINKED ISSUE #2404: [REPOSITORY REQUEST] XML Security Library
Hello masters,

It would be nice to publish the XML Security Library in the MinGW-w64 packages.

**Description from [its official site](https://www.aleksey.com/xmlsec):**

> XML Security Library is a C library based on [LibXML2](http://xmlsoft.org/). The library supports major XML security standards:
> 
> [XML Signature](http://www.w3.org/TR/xmldsig-core)
> [XML Encryption](http://www.w3.org/TR/xmlenc-core/)
> [Canonical XML](http://www.w3.org/TR/xml-c14n) (part of the [LibXML2](http://xmlsoft.org/))
> [Exclusive Canonical XML](http://www.w3.org/TR/xml-exc-c14n) (part of the [LibXML2](http://xmlsoft.org/))
> XML Security Library is released under the MIT Licence see the Copyright file in the distribution for details.

**Latest release and news:**

> April 20 2017
>
> The [XML Security Library 1.2.24](https://www.aleksey.com/xmlsec/download.html) release includes the following changes:
> - Added ECDSA-SHA1, ECDSA-SHA256, ECDSA-SHA512 support for xmlsec-nss (vmiklos).
> - Fixed XMLSEC_KEYINFO_FLAGS_X509DATA_DONT_VERIFY_CERTS handling (vmiklos).
> - Disabled external entities loading by xmlsec utility app by default to prevent XXE attacks (d-hat).
> - Improved OpenSSL version and features detection.
> - Cleaned up, simplified, and standardized internal error reporting.
> - Fixed a few Coverity-discovered bugs ([report](https://scan.coverity.com/projects/xmlsec)).
> - Marked as deprecated all the functions in xmlsec/soap.h file and a couple other functions no longer required by xmlsec. These functions will be removed in the future releases.
> - Several other small fixes ([more details](https://github.com/lsh123/xmlsec/commits/master)).
> - Please note that OpenSSL 0.9.8 support will be removed in the next release of XMLSec library.

Thank you!

### PR DESCRIPTION
**Pull request by peterbud**
All tests except one were running successfully.

### CODE REVIEW

**Comment by mati865** [comment]
sed: can't read ./xmlsec1-gnutls.pc: No such file or directory
==> ERROR: A failure occurred in package().

**Comment by peterbud** [comment]
Strange. It is creating the file [here](https://ci.appveyor.com/project/Alexpux/mingw-packages/build/1.0.2846#L405), and it works locally.
Let me see what is wrong here....

File: mingw-w64-xmlsec/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+build() {
+    [[ -d ${srcdir}/build-${MINGW_CHOST} ]] && rm -rf ${srcdir}/build-${MINGW_CHOST}
+    mkdir -p ${srcdir}/build-${MINGW_CHOST} && cd ${srcdir}/build-${MINGW_CHOST}
+
+    ../${_realname}-${_realname}-${pkgver//./_}/configure \
+        --prefix="${MINGW_PREFIX}"      \
+        --build=${MINGW_CHOST}          \
+        --host=${MINGW_CHOST}           \
```
Comment (Alexpux) [inline mingw-w64-xmlsec/PKGBUILD]: Remove extra slash

File: mingw-w64-xmlsec/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Maintainer: Peter Budai <peterbudai at hotmail dot com>
+
+_realname=xmlsec
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=1.2.24
+pkgrel=1
+pkgdesc="XML Security Library is a C library based on LibXML2. The library supports major XML security standards. (mingw-w64)"
```
Comment (Alexpux) [inline mingw-w64-xmlsec/PKGBUILD]: Remove second sentence and last dot before (mingw-w64)

**Comment by Alexpux** [comment]
@peterbud fix my comments

**Pull request merged**